common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -1867,7 +1867,7 @@ public static enum ConfVars {
             + "execution engine doesn't take advantage of statistics stored in the cache."),
     // CBO related
     HIVE_CBO_ENABLED("hive.cbo.enable", true, "Flag to control enabling Cost Based Optimizations using Calcite framework."),
-    HIVE_CBO_FALLBACK_STRATEGY("hive.cbo.fallback.strategy", "CONSERVATIVE",
+    HIVE_CBO_FALLBACK_STRATEGY("hive.cbo.fallback.strategy", "NEVER",
```

Comment:
Can we make NEVER default? I guess this is the last known issue and that's why this PR would change it. It is one of the greatest news in 2024.
https://issues.apache.org/jira/browse/HIVE-27831

I'd create a separate ticket to change the default, cleaning up some test resources as [Stamatis tried before](https://github.com/apache/hive/pull/4836/files).